x-pack/plugins/ux/public/services/data/environments_query.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ESSearchResponse } from '@kbn/core/types/elasticsearch';
+import {
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../common/environment_filter_values';
+import { Environment } from '../../../common/environment_rt';
+
+export function transformEnvironmentsResponse<T>(
+  response?: ESSearchResponse<T, ReturnType<typeof getEnvironments>>
+) {
+  if (!response) return response;
+
+  const aggs = response.aggregations;
+  const environmentsBuckets = aggs?.environments.buckets || [];
+
+  const environments = environmentsBuckets.map(
+    (environmentBucket) => environmentBucket.key as string
+  );
+
+  return environments as Environment[];
+}
+
+export function getEnvironments({
+  serviceName,
+  size,
+  start,
+  end,
+}: {
+  serviceName?: string;
+  size: number;
+  start: number;
+  end: number;
+}) {
+  return {
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
```

Comment:
i think we should also add a filter for filtering out the rum data, add a filter for transaction.type page-load

Reply:
Does https://github.com/elastic/kibana/pull/134393/commits/159a1273d3080df3c723a6ad4f8f7d51b5d8af27 look ok to you?